src/core/utils/types/ids.ts

```diff
@@ -1,3 +1,4 @@
 export type FaustId = `${number}`;
 export type Pid = `${number}-${string}:${FaustId}`;
 export type WorkId = `work-of:${number}-${string}:${FaustId}`;
+export type App = "material" | "search-header" | "search-result";
```

Comment:
Consider if loan list should be included in this list. You could also rename it to only contain ids for guarded apps.

Reply:
I have removed the search header since it has no guarded requests.
This should have a small effect on performance as well. If it is measurable I don't know.